File: dotnet/src/Core/Models/Entities/Enum.cs
namespace Agience.Core.Models.Entities
{
    public enum Visibility
    {
        Private = 0,
        Public = 1
    }

    public enum AuthorizationType
    {
        None = 0,
        OAuth2 = 1,
        ApiKey = 2
    }

    public enum PluginType
    {
        Curated = 0,
        Compiled = 1
    }

    public enum CompletionAction
    {
        Idle = 0,
        Restart = 1        
    }
}

File: python/src/core/models/entities/enum.py
from enum import Enum, auto


class Visibility(Enum):
    Private = 0
    Public = 1


class AuthorizationType(Enum):
    None_ = 0  # Using None_ since 'None' is reserved in Python
    OAuth2 = 1
    ApiKey = 2


class PluginType(Enum):
    Curated = 0
    Compiled = 1


class CompletionAction(Enum):
    Idle = 0
    Restart = 1
